### ISSUE TITLE
generate tiles does not stride correctly at level>0

### ISSUE THREAD

**Issue by ralfbanisch** [Dana-Farber-AIOS/pathml#240]
**Describe the bug**
The `generate_tiles` function of the openslide backend (https://github.com/Dana-Farber-AIOS/pathml/blob/6e4208736f29bd38143368b0ce8edc8be9b20fa3/pathml/core/slide_backends.py#L137) does not stride correctly when a level other than zero is selected.

**To Reproduce**
installed release 1.0.3.

```
from pathml.core import HESlide
from pathml.preprocessing import Pipeline, BoxBlur, TissueDetectionHE

level = 5
tile_size = 512
wsi = HESlide(sample_wsi_path)

pipeline = Pipeline([
    TissueDetectionHE(mask_name = "tissue", outer_contours_only=True)
])

wsi.run(pipeline, tile_size=tile_size, level=level, overwrite_existing_tiles=True, distributed=False)
for tile in wsi.tiles:
    print(tile)
```

**Expected behavior**
Since the stride keyword was not passed, I expect non-overlapping tiles. However, this is not what happens.

The issue is that the `location` argument of `openslide.read_region()` refers to the base coordinate system. For non-overlapping tiles the tile coords need to be strided by `tile_size*slide.level_downsamples[level]`. However, they are only strided by `tile_size`, so effectively only highly overlapping tiles from the top left corner of the wsi are extracted (in my case, all these tiles are background).

**Comment by jacobrosenthaldfci** [comment]
hi @ralfbanisch , thanks for uncovering this!!
I just opened a PR that I hope will fix this (#242) by scaling the coordinates in `OpenSlideBackend.extract_region()` before passing them to `openslide.read_region()`. The only minor comment is that I think they need to be scaled by sqrt(slide.level_downsamples[level]) because the level_downsamples is the product of the downsampling factor in both x and y dimensions.

Thanks for contributing!!

**Comment by ralfbanisch** [comment]
Hi @jacobrosenthaldfci, are you sure about the `sqrt`? I just tested this locally and I get the same region back with these two commands:

```
wsi = HESlide(sample_wsi_path)

location = (33000, 61000)
# size of the region in (width, height)
size = (2000, 2000)
region = wsi.slide.extract_region(location = location, size = size, level=0)
level = 1
scale = wsi.slide.slide.level_downsamples[level]
new_size = (int(size[0] / scale), int(size[1] / scale))
region = wsi.slide.extract_region(location = location, size = new_size, level=level)
plt.imshow(region)
plt.axis('off')
plt.show()
```
That seems to suggest that `slide.level_downsamples` is the scaling factor in each dimension. That's also how I always understood it.

**Comment by jacobrosenthaldfci** [comment]
Yes, you are absolutely right, my mistake!

I just fixed and pushed to #242, here's what I did to make sure that it really works this time:
````python
wsi = HESlide("data/CMU-1.svs")

tile_size = 2000
i, j = wsi.slide.get_image_shape(level = 1)

fig, axarr = plt.subplots(nrows = i // tile_size + 1, ncols = j // tile_size + 1)

for tile in wsi.generate_tiles(shape = tile_size, level = 1, pad = True):
    row_ix = int(tile.coords[0] / tile_size)
    col_ix = int(tile.coords[1] / tile_size)
    axarr[row_ix, col_ix].imshow(tile.image)

for ax in axarr.ravel():
    ax.axis("off")

plt.tight_layout()
plt.show()
````
![image](https://user-images.githubusercontent.com/53529525/144079173-dd33fb9a-e1a9-4fa1-abf4-d02ed16d3296.png)

Let me know what you think, thanks so much!
